test/test_estimator.py

```diff
@@ -36,27 +36,26 @@
 DATAOBJ_SIZE = (5, 5, 5, 4)
 
 
-class DummyModel(BaseModel):
-    def __init__(self, **kwargs):
-        self._rng = np.random.default_rng(1234)
-        super().__init__(**kwargs)
+class DummyInsiderModel(BaseModel):
+    def __init__(self, dataset, **kwargs):
+        super().__init__(dataset, **kwargs)
 
     def fit_predict(self, index: int | None = None, **kwargs):
-        # Return a synthetic 3D image
-        return self._rng.random(DATAOBJ_SIZE[:-1])
+        # Return the indexed volume
+        return self._dataset.dataobj[..., index]
```

Comment:
I could be wrong, but I believe `fit_predict` is never called because of the iterator mock.